src/js/tabs/motors.js

```diff
@@ -678,33 +678,35 @@ TABS.motors.initialize = function (callback) {
                 self.analyticsChanges['BidirectionalDshot'] = newValue;
                 FC.MOTOR_CONFIG.use_dshot_telemetry = value;
 
-                FC.FILTER_CONFIG.dyn_notch_count = self.previousFilterDynCount;
-                FC.FILTER_CONFIG.dyn_notch_q = self.previousFilterDynQ;
-
-                const dialogSettings = {
-                    title: i18n.getMessage("dialogDynFiltersChangeTitle"),
-                    text: i18n.getMessage("dialogDynFiltersChangeNote"),
-                    buttonYesText: i18n.getMessage("presetsWarningDialogYesButton"),
-                    buttonNoText: i18n.getMessage("presetsWarningDialogNoButton"),
-                    buttonYesCallback: () => _dynFilterChange(),
-                    buttonNoCallback:  null,
-                };
-
-                const _dynFilterChange = function() {
-                    if (value && !self.previousDshotBidir) {
-                        FC.FILTER_CONFIG.dyn_notch_count = FILTER_DEFAULT.dyn_notch_count_rpm;
-                        FC.FILTER_CONFIG.dyn_notch_q = FILTER_DEFAULT.dyn_notch_q_rpm;
-                    } else if (!value && self.previousDshotBidir) {
-                        FC.FILTER_CONFIG.dyn_notch_count = FILTER_DEFAULT.dyn_notch_count;
-                        FC.FILTER_CONFIG.dyn_notch_q = FILTER_DEFAULT.dyn_notch_q;
-                    }
-                };
-
-                if (FC.MOTOR_CONFIG.use_dshot_telemetry !== self.previousDshotBidir) { // if rpmFilterEnabled is not the same value as saved in the fc
-                    GUI.showYesNoDialog(dialogSettings);
-                } else {
+                if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_44)) {
                     FC.FILTER_CONFIG.dyn_notch_count = self.previousFilterDynCount;
                     FC.FILTER_CONFIG.dyn_notch_q = self.previousFilterDynQ;
+
+                    const dialogSettings = {
+                        title: i18n.getMessage("dialogDynFiltersChangeTitle"),
+                        text: i18n.getMessage("dialogDynFiltersChangeNote"),
+                        buttonYesText: i18n.getMessage("presetsWarningDialogYesButton"),
+                        buttonNoText: i18n.getMessage("presetsWarningDialogNoButton"),
+                        buttonYesCallback: () => _dynFilterChange(),
+                        buttonNoCallback:  null,
+                    };
+
+                    const _dynFilterChange = function() {
```

Comment:
Can we put this function in `fc.js` as it used in two tabs now? Including `dialog[DynFilter]Settings`
But is up to you to decide.

Reply:
I think fc.js only have fc related settings, not a good place.